build_runner/lib/src/asset_graph/graph.dart

```diff
@@ -336,3 +343,29 @@ class AssetGraph {
   void add(AssetNode node) => _add(node);
   Set<AssetId> remove(AssetId id) => _removeRecursive(id);
 }
+
+/// Computes a [Digest] for [buildActions] which can be used to compare one set
+/// of [BuildAction]s against another.
+Digest computeBuildActionsDigest(Iterable<BuildAction> buildActions) {
```

Comment:
private?

Reply:
This is used from build_definition as well, I could move it elsewhere?

Reply:
We could also pass the list of build actions into the `deserialize` method and do the validation there but it feels weird.

Reply:
ah fair enough. I'm fine if it's public then.